source/shared/core_stmt.cpp

```diff
@@ -3223,6 +2128,986 @@ int round_up_decimal_numbers(_Inout_ char* buffer, _In_ int decimal_pos, _In_ in
+        // original allocation null terminates the buffer already.
+        if (str_len < buffer_length) {
+            str[str_len] = '\0';
+        }
+        core::sqlsrv_zval_stringl(value_z, str, str_len);
+    }
+    else {
+        if (encoding != SQLSRV_ENCODING_CHAR) {
+            char* outString = NULL;
+            SQLLEN outLen = 0;
+
+            bool result = convert_string_from_utf16(encoding, reinterpret_cast<const SQLWCHAR*>(str), int(str_len / sizeof(SQLWCHAR)), &outString, outLen);
+            CHECK_CUSTOM_ERROR(!result, stmt, SQLSRV_ERROR_OUTPUT_PARAM_ENCODING_TRANSLATE, get_last_error_message()) {
+                throw core::CoreException();
+            }
+
+            if (stmt->format_decimals && (sql_data_type == SQL_DECIMAL || sql_data_type == SQL_NUMERIC)) {
+                format_decimal_numbers(NO_CHANGE_DECIMAL_PLACES, decimal_digits, outString, &outLen);
+            }
+
+            core::sqlsrv_zval_stringl(value_z, outString, outLen);
+            sqlsrv_free(outString);
+        }
+        else {
+            if (stmt->format_decimals && (sql_data_type == SQL_DECIMAL || sql_data_type == SQL_NUMERIC)) {
+                format_decimal_numbers(NO_CHANGE_DECIMAL_PLACES, decimal_digits, str, &str_len);
+            }
+
+            core::sqlsrv_zval_stringl(value_z, str, str_len);
+        }
+    }
+
+    value_z = NULL;
+}
+
+void sqlsrv_param_inout::resize_output_buffer(_Inout_ zval* param_z, _In_ bool is_numeric)
+{
+    // Prerequisites: buffer, buffer_length, column_size, and strlen_or_indptr have been set to a known value 
+    // Purpose: 
+    // Verify there is enough space to hold the output string parameter, and allocate if necessary. The param_z
+    // is updated to contain the new buffer with the correct size and its reference is incremented, and all required 
+    // values for SQLBindParameter will also be updated. 
+    SQLLEN original_len = buffer_length;
+    SQLLEN expected_len;
+    SQLLEN buffer_null_extra;
+    SQLLEN elem_size;
+
+    // Calculate the size of each 'element' represented by column_size.  WCHAR is the size of a wide char (2), and so is 
+    // a N(VAR)CHAR/NTEXT field being returned as a binary field.
+    elem_size = (c_data_type == SQL_C_WCHAR || 
+                (c_data_type == SQL_C_BINARY && 
+                    (sql_data_type == SQL_WCHAR || sql_data_type == SQL_WVARCHAR || sql_data_type == SQL_WLONGVARCHAR))) ? sizeof(SQLWCHAR) : sizeof(SQLCHAR);
+
+    // account for the NULL terminator returned by ODBC and needed by Zend to avoid a "String not null terminated" debug warning
+    SQLULEN field_size = column_size;
+
+    // With AE enabled, column_size is already retrieved from SQLDescribeParam, but column_size
+    // does not include the negative sign or decimal place for numeric values
+    // VSO Bug 2913: without AE, the same can happen as well, in particular to decimals
```

Comment:
Is this still a bug or are we able to address it now?

Reply:
This VSO bug is already resolved. It was kept because the issue contains details, but of course I can remove it.